pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -779,3 +811,61 @@ func (c *Cluster) applyUpgradeOSDFunctionality() {
 		}
 	}
 }
+
+// replaceOSDForNewStore deletes an existing OSD deployment that does not not match the expected OSD backend store provided in the storage spec
+func (c *Cluster) replaceOSDForNewStore() error {
+	if c.spec.Storage.Store.UpdateStore != OSDStoreUpdateConfirmation {
+		logger.Debugf("no OSD migration to a new backend store is requested")
+		return nil
+	}
+
+	osdToReplace, err := c.getOSDReplaceInfo()
+	if err != nil {
+		return errors.Wrapf(err, "failed to get OSD replace info")
+	}
+
+	if osdToReplace == nil {
+		logger.Info("no osd to replace")
+		return nil
+	}
+
+	logger.Infof("starting migration of the OSD.%d", osdToReplace.ID)
+
+	// Delete the OSD deployment
+	deploymentName := fmt.Sprintf("rook-ceph-osd-%d", osdToReplace.ID)
+	logger.Infof("removing the OSD deployment %q", deploymentName)
+	if err := k8sutil.DeleteDeployment(c.clusterInfo.Context, c.context.Clientset, c.clusterInfo.Namespace, deploymentName); err != nil {
+		if err != nil {
+			if kerrors.IsNotFound(err) {
+				logger.Debugf("osd deployment %q not found. Ignoring since object must be deleted.", deploymentName)
+			} else {
+				return errors.Wrapf(err, "failed to delete OSD deployment %q.", deploymentName)
+			}
+		}
+	}
+
+	c.replaceOSD = osdToReplace
+
+	return osdToReplace.saveAsConfig(c.context, c.clusterInfo)
+}
+
+func (c *Cluster) waitForHealthyPGs() (bool, error) {
+	waitFunc := func() (done bool, err error) {
+		pgHealthMsg, pgClean, err := cephclient.IsClusterClean(c.context, c.clusterInfo)
+		if err != nil {
+			return false, errors.Wrap(err, "failed to check pg are healthy")
+		}
+		if pgClean {
+			return true, nil
+		}
+		logger.Warningf("pgs are not healthy. PG status: %q", pgHealthMsg)
```

Comment:
```suggestion
		logger.Infof("waiting for PGs to be healthy after replacing an OSD, status: %q", pgHealthMsg)
```

Reply:
Can we also just print this every minute or two instead of every 10s?

Reply:
I'll take this up in a follow PR next week.